doc/releases: add release notes for nautilus v14.2.5

smithfarm opened the pull request.

theanalyst commented on doc/releases/nautilus.rst:
```diff
@@ -1,3 +1,361 @@
+
+* New OSD daemon command dump_scrub_reservations which reveals the
+  scrub reservations that are held for local (primary) and remote (replica) PGs.
+
```
A couple of RGW backports are worth mentioning. (cc @cbodley)
- RGW now supports S3 Object Lock set of APIs allowing for a WORM model for storing objects. 6 new APIs have been added put/get bucket object lock, put/get object retention, put/get object legal hold.
- RGW now supports List Objects V2

emmericp replied:
+1 on mentioning the object lock feature prominently, it's a pretty important feature for backup use cases

liewegas commented on doc/releases/nautilus.rst:
```diff
@@ -1,3 +1,361 @@
+
+Notable Changes
+---------------
+
```
This release fixes a critical BlueStore bug (link) introduced in 14.2.3 (and also present in 14.2.4) that can lead to data corruption when a separate 'wal' device is used.

liewegas replied:
@ifed01 what else should we say here?

ifed01 replied:
@liewegas - IMO this is good enough.

neha-ojha replied:
> This release fixes a critical BlueStore bug (link) introduced in 14.2.3 (and also present in 14.2.4) that can lead to data corruption when a separate 'wal' device is used.

https://tracker.ceph.com/issues/42223

neha-ojha commented:
@dzafman found an issue https://tracker.ceph.com/issues/43124 with https://github.com/ceph/ceph/pull/30899/commits/d6b3025a279c3876df035b02c7b8ec13e0590d7f, which removes all upmaps for crushmaps with more than one choose step before an emit, causing imbalance in PGs and unnecessary data movement. We'd like to pull this commit out https://github.com/ceph/ceph/pull/32018 before releasing. 
FYI @liewegas @smithfarm @djgalloway @theanalyst @yuriw

smithfarm commented:
@theanalyst @liewegas Please take another look?

smithfarm commented:
Changelog:

* dropped reference to commit that is being reverted

smithfarm commented:
@neha-ojha I generally do not mention reverts in the changelog - instead, I remove the mention of the commit(s) that were reverted. See https://github.com/ceph/ceph/pull/31970#issuecomment-562227320

smithfarm commented:
Someone please merge this as soon as it becomes possible to do so.

liewegas commented on doc/releases/nautilus.rst:
```diff
@@ -1,3 +1,382 @@
+  corruption when a separate "WAL" device is used.
+
+New health warnings:
+
```
* Ceph will now issue health warnings if daemons have recently crashed.  Ceph has been collecting crash reports since the initial Nautilus release, but the health alerts are new.  To view new crashes (or all crashes, if you've just upgraded),::

ceph crash ls-new

To acknowledge a particular crash (or all crashes) and silence the health warning,::

ceph crash archive <crash-id>
ceph crash archive-all

smithfarm replied:
OK, added it.

smithfarm commented:
Changelog:

* added new release note suggested by Sage
* fixed a typo ("as" should have been "has")
* some more, relatively minor, language finessing

neha-ojha commented:
@smithfarm @djgalloway We are also including https://github.com/ceph/ceph/pull/32045

smithfarm commented:
Changelog:

* mention https://github.com/ceph/ceph/pull/32045 in the release notes
* in the Changelog section, add missing `* ` at beginning of each line

mykaul commented on doc/releases/nautilus.rst:
```diff
@@ -1,3 +1,395 @@
+* The telemetry module now has a 'device' channel, enabled by default, that
+  will report anonymized hard disk and SSD health metrics to telemetry.ceph.com
+  in order to build and improve device failure prediction algorithms. Because
+  the content of telemetry reports has changed, you will need to either re-opt-in
```
'either' is not needed here.

smithfarm commented:
Changes since the last force-push:

* mention #30382 in Changelog
* mention #30394 in Changelog

smithfarm commented:
Changes in the most recent force-push:

* drop superfluous word `either` - thanks, @mykaul

djgalloway commented:
> @smithfarm @djgalloway We are also including #32045

:disappointed: The build had already started.  Luckily (?) arm64 on centos7 didn't finish so I can rebuild and retag once @liewegas gives me the go-ahead.

smithfarm commented:
Changes since last force-push:

added

```
* mgr/devicehealth: import _strptime directly (`pr#32082 <https://github.com/ceph/ceph/pull/32082>`_, Sage Weil)
```

theanalyst commented:
jenkins render docs

ceph-jenkins commented:
Doc render available at http://docs.ceph.com/ceph-prs/31970/

smithfarm commented:
@liewegas Since v14.2.5 is now out, could you take another look at this and revisit your change request? I can't merge it...

smithfarm commented:
jenkins render docs

smithfarm commented:
@liewegas Regarding the two "telemetry" bullet points, the language about re-opting-in is repeated in both. Is that intentional? Would it be worth considering to merge the first telemetry bullet point (which seems specific to "device") with the second one ("basic" == "all the new info")?

ceph-jenkins commented:
Doc render available at http://docs.ceph.com/ceph-prs/31970/
